server/asset/dcr/utxo.go

```diff
@@ -381,7 +379,10 @@ func auditContract(op *Output) (*asset.Contract, error) {
 		return nil, fmt.Errorf("invalid script version %d", op.scriptVersion)
 	}
 	output := tx.outs[int(op.vout)]
-	scriptHash := dexdcr.ExtractScriptHash(output.pkScript)
+	if output.version != 0 {
+		return nil, fmt.Errorf("unsupported script version %d", output.version)
+	}
+	scriptHash := stdscript.ExtractScriptHashV0(output.pkScript)
```

Comment:
I see 
```
if op.scriptVersion != 0 {
		return nil, fmt.Errorf("invalid script version %d", op.scriptVersion)
}
```
right above this, is it the same check?

Also, something I've been wondering is if we should also verify the transaction version is 1? 

I believe we are already using one exclusively. There are three possible at the moment, 1, the initial or default we use, 2 enables sequence locks, which I'm not sure but could be used to make a transaction that we think is fine when examining but cannot be mined because of the sequence? and three enables treasury opcodes, I believe. So maybe forcing one also prevents funding with treasury gens?

If there was a new version in the future that allows something that breaks the dex, we would want to prevent that here. I think.

Reply:
Good spot.  It's actually quite peculiar that this function is using `op.tx.outs` at all.  Both the version and the pkScript itself are fields of the `Output`...  The double check of version in the two different places is alright, but weird.  Not sure if I wanna change it yet.

Good question about the tx version.  Perhaps we should make an issue to get that sorted with outside input.